js/src/redux/providers/registry/middleware.js

```diff
@@ -14,13 +14,36 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
+import { debounce } from 'lodash';
+import store from 'store';
 import Contracts from '~/contracts';
 import subscribeToEvents from '~/util/subscribe-to-events';
 
 import registryABI from '~/contracts/abi/registry.json';
 
 import { setReverse, startCachingReverses } from './actions';
 
+const STORE_KEY = '_parity::reverses';
+
+const read = (chain) => {
+  const reverses = store.get(`${STORE_KEY}::${chain}::data`);
```

Comment:
Would probably make sense to const-ify `data` & `lastBlock` since it is used in multiple places. (Not a crisis either way)